src/EdFi.AnalyticsMiddleTier.Tests/Dimensions/When_querying_the_StudentHistoryDim_view.cs

```diff
@@ -0,0 +1,189 @@
+            {
+                (bool success, string errorMessage) testResult = DataStandard.RunTestCase<StudentHistoryDim>($"{TestCasesFolder}.{_caseIdentifier}_should_have_StudentKey.xml");
+                testResult.success.ShouldBe(true, testResult.errorMessage);
+            }
+
+            [Test]
+            public void Then_should_have_StudentSchoolKey()
+            {
+                (bool success, string errorMessage) testResult = DataStandard.RunTestCase<StudentHistoryDim>($"{TestCasesFolder}.{_caseIdentifier}_should_have_StudentSchoolKey.xml");
+                testResult.success.ShouldBe(true, testResult.errorMessage);
+            }
+
+            [Test]
+            public void Then_should_have_LastModifiedDate()
+            {
+                (bool success, string errorMessage) testResult = DataStandard.RunTestCase<StudentHistoryDim>($"{TestCasesFolder}.{_caseIdentifier}_should_have_LastModifiedDate.xml");
+                testResult.success.ShouldBe(true, testResult.errorMessage);
+            }
+
+            [Test]
+            public void Then_should_have_GradeSummary()
+            {
+                (bool success, string errorMessage) testResult = DataStandard.RunTestCase<StudentHistoryDim>($"{TestCasesFolder}.{_caseIdentifier}_should_have_GradeSummary.xml");
+                testResult.success.ShouldBe(true, testResult.errorMessage);
+            }
+            [Test]
+            public void Then_should_have_CurrentSchoolKey()
+            {
+                (bool success, string errorMessage) testResult = DataStandard.RunTestCase<StudentHistoryDim>($"{TestCasesFolder}.{_caseIdentifier}_should_have_CurrentSchoolKey.xml");
+                testResult.success.ShouldBe(true, testResult.errorMessage);
+            }
+            [Test]
+            public void Then_should_have_AttendanceRate()
+            {
+                (bool success, string errorMessage) testResult = DataStandard.RunTestCase<StudentHistoryDim>($"{TestCasesFolder}.{_caseIdentifier}_should_have_AttendanceRate.xml");
+                testResult.success.ShouldBe(true, testResult.errorMessage);
+            }
+            [Test]
+            public void Then_should_have_ReferralsAndSuspensions()
+            {
+                (bool success, string errorMessage) testResult = DataStandard.RunTestCase<StudentHistoryDim>($"{TestCasesFolder}.{_caseIdentifier}_should_have_ReferralsAndSuspensions.xml");
+                testResult.success.ShouldBe(true, testResult.errorMessage);
+            }
+            [Test]
+            public void Then_should_have_EnrollmentHistory()
+            {
+                (bool success, string errorMessage) testResult = DataStandard.RunTestCase<StudentHistoryDim>($"{TestCasesFolder}.{_caseIdentifier}_should_have_EnrollmentHistory.xml");
+                testResult.success.ShouldBe(true, testResult.errorMessage);
+            }
+        }
+
+        public class Given_StudentHistoryDim_224082_867530023
```

Comment:
Unlike the test case above, this one does not have assertions for StudentKey and CurrentSchoolKey and the data inserted for this test and the one above do not look very different between the two, so I am unsure of the distinction between these two test cases. Can you explain what the differences are?

Reply:
@bptillman  In this test case we added rows to verify additional conditions related to subqueries (AttendanceRate, EnrollmentHistory, GradeSummary and Referrals and Suspensions) and the max LastModifiedDate. Those fields could be affected by changes in our query, StudentKey and CurrentSchoolKey conditions can be validated in the first test case class. Do you want test cases for those conditions (Student and CurrentSchool)?

Reply:
That makes sense to me. So no need to update the tests cases then. I will approve this PR then since all other testing I have done appears to be working.